lib/dns.js

```diff
@@ -311,26 +318,31 @@ function setServers(servers) {
   const orig = cares.getServers();
   const newSet = [];
   const IPv6RE = /\[(.*)\]/;
-  const addrSplitRE = /:\d+$/;
+  const addrSplitRE = /(^.+?)(?::(\d+))?$/;
```

Comment:
another possible way to handle this that may provide more robust and consistent handling of the address and port would be to leverage the new `URL` parser. Essentially:

```js
const URL = require('url').URL;
const u = new URL(`dns://${serv}`);
```

If perf is a concern around creating multiple `URL` objects, one can be created and reused:

```js
const url = new URL(`dns://${serv}`);
// use it
url.host = nextServ;
// etc
```